### ISSUE TITLE
envdiff fails when "options.envdiff.ignored_keys" is set

### ISSUE THREAD

**Issue by LeShaunJ** [nickthecook/crops#6]
## Symptoms

For instance:

> ```yaml
> # ops.yml
> options:
>   envdiff:
>     ignored_keys:
>       - TF_VAR_something
>   exec:
> ```
> 
> Fails with:
> 
> ```sh
> ops envdiff test staging
> ```
> ```
> Unhandled exception: Option 'envdiff.ignored_keys' must be an array of strings; got YAML::Any. (Options::OptionsError)
>   from /Users/username/.gem/ruby/2.7.4/gems/ops_team-2.0.6.rc1/build/darwin_arm64/ops in 'raise<Options::OptionsError>:NoReturn'
>   from /Users/username/.gem/ruby/2.7.4/gems/ops_team-2.0.6.rc1/build/darwin_arm64/ops in 'Builtins::Envdiff#keys_for<String>:Array(String)'
>   from /Users/username/.gem/ruby/2.7.4/gems/ops_team-2.0.6.rc1/build/darwin_arm64/ops in 'Builtins::Envdiff#source_only_keys:Array(String)'
>   from /Users/username/.gem/ruby/2.7.4/gems/ops_team-2.0.6.rc1/build/darwin_arm64/ops in 'Ops#run:(Bool | Nil)'
>   from /Users/username/.gem/ruby/2.7.4/gems/ops_team-2.0.6.rc1/build/darwin_arm64/ops in '__crystal_main'
>   from /Users/username/.gem/ruby/2.7.4/gems/ops_team-2.0.6.rc1/build/darwin_arm64/ops in 'main'
> ```

Meanwhile:

> ```yaml
> # ops.yml
> options:
>   # envdiff:
>     # ignored_keys:
>       # - TF_VAR_something
>   exec:
> ```
> 
> Is business as usual:
> 
> ```sh
> ops envdiff test staging
> ```
> ```
> Environment 'test' defines keys that 'staging' does not:
> 
>    - [CONFIG] TF_VAR_something
> 
> Environment 'staging' defines keys that 'test' does not:
> 
>    - [CONFIG] TF_VAR_something_else
> ```

## Diagnosis

From what I can see, `Builtins::Envdiff` is doing it's job:
https://github.com/nickthecook/crops/blob/f4cfdadea96886a2edd76e45cce83b51859bb9da/src/builtins/envdiff.cr#L127-L132

This, due to the nature of `Options#get`, as there's no way to know what type the specified option may be at compile time; hence, `YAML::Any | Nil` makes sense as it's return type:
https://github.com/nickthecook/crops/blob/f4cfdadea96886a2edd76e45cce83b51859bb9da/src/options.cr#L9-L12

## Suggestion

Off the top of my head (_and  limited `crystal` knowledge_), something like the following _may_ do the trick:
```crystal
# crops/src/builtins/envdiff.cr
module Builtins
    class Envdiff < Builtin
        # ...
        private def ignored_keys : Array(String)
            @ignored_keys ||= begin
# -             option = Options.get("envdiff.ignored_keys")
                option = Options.get("envdiff.ignored_keys").as?(Array(String))

                if option.nil?
                    [] of String
                else
                    raise Options::OptionsError.new("Option 'envdiff.ignored_keys' must be an array of strings; got #{option.class.name}.") unless option.is_a?(Array(String))

                    option
                end
            end
        end
    end
end
```
From the `crystal` [docs](https://crystal-lang.org/reference/1.8/syntax_and_semantics/as_question.html):

> The `as?` pseudo-method is similar to `as`, except that it returns `nil` instead of raising an exception when the type doesn't match. It also can't be used to cast between pointer types and other types.

**Comment by nickthecook** [comment]
Thanks for the detailed report!

I will take a look.

**Comment by nickthecook** [comment]
Fixed in 2.0.8.
